app/controllers/admin/invitations_controller.rb

```diff
@@ -13,17 +13,15 @@ def new; end
 
     def create
       email = params.dig(:user, :email)
-      name = params.dig(:user, :name)
 
       if User.exists?(email: email.downcase, registered: true)
         flash[:error] = "Invitation was not sent. There is already a registered user with the email: #{email}"
         redirect_to admin_invitations_path
         return
       end
 
-      username = "#{name.downcase.tr(' ', '_').gsub(/[^0-9a-z ]/i, '')}_#{rand(1000)}"
+      username = "#{email.split('@').first.gsub(/[^0-9a-z ]/i, '')}_#{rand(1000)}"
```

Comment:
should we downcase this before searching for the class a-z? Local part of email addresses are case sensitive, and a very realistic email address might look like Dan.Uber@gmail.com (the fact that google's servers are case insensitive and ignore period/dot characters actually makes this worse, since you can have admins inviting any mix of cases, and it gets delivered independently of the mailbox name on the host)

I think I mean this:

```ruby
      username = "#{email.split('@').first.downcase.gsub(/[^0-9a-z ]/i, '')}_#{rand(1000)}"
```

Reply:
oh - it looks like email gets downcased during user creation - ignore my comment!